File: TP2_6216948_PW/TP2_6216948_PWS/Models/Saison.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TP2_6216948_PWS.Models
{
    public class Saison
    {
        [Key]
        public int Id { get; set; }

        public string DateDebut { get; set; }

        public string DateFin { get; set; }

        public int Annee { get; set; }




        //prop nav
        [ForeignKey("League")]
        public int LeagueId { get; set; }

        public virtual League League { get; set; }


    }
}

File: TP2_6216948_Angular/src/Models/Saison.ts
import { League } from "./League";

export class Saison{
    constructor(
        
        public id:number,
        public dateDebut:string,
        public dateFin:string,
        public annee:number,

        public leagueId?:number,
        public league?:League,

    ){

    }
}